internal/templates/templatesfunctions.go

```diff
@@ -214,6 +216,9 @@ var (
 		"month": func(month int) string {
 			return gametime.MonthName(month)
 		},
+		"t": func(msgID string, tplData ...map[string]string) string {
+			return T(msgID, tplData...)
+		},
```

Comment:
i'm curious, how do we handle cases where there are 1 or more variables in the text such as:

David eats `1` `apple`.

I would think something like:

**English/Default:**
David eats `%[1]v` `%[2]v`.

**Translated:**
デイビッドは`%[2]v`を`%[1]v`個食べる

**Template**
{{  t ``David eats %[1]v %[2]v.`` qty, item.Name }}

In the above example, the number and noun order are flipped, but a `Sprintf` can be performed with ordered values. Maybe another function name for this scenario, like `argt`

Reply:
Looking a bit more, it seems we would use the map argument to define template variables that would be substituted... so:

**English/Default:**
David eats `{{ .qty }}` `{{ .name }}`.

**Translated:**
デイビッドは`{{ .name }}`を`{{ .qty }}`個食べる

I am leaving my original question/comment in case I am mistaken.

**Template**
{{ t "David eats `{{ .qty }}` `{{ .name }}`."  }}

---

How is the map of translation values created/passed to the `t` function?

Reply:
Yes, using golang's template syntax.

For arguments passing, it is more convenient to handle `templates.T` function in `.go` files, but in the `t` function of the template, it will be a bit troublesome, because it needs to change the parsing logic in `templates.Process`, and the impact of the change may be large, so need to reconsider the proper way of implementation.

Reply:
Add a template function `map` for translate function argument parsing, example:

tempalte:
```
{{- $map := (map "ColumnCount" .ColumnCount) -}}
{{ t "Table Column Count: {{ .ColumnCount }}" $map }}
```

localize file(`zh.yaml`):

```
'Table Column Count: {{ .ColumnCount }}': '表格列数: {{ .ColumnCount }}'
```

Screenshot:
![Screenshot1](https://github.com/user-attachments/assets/da649b5e-53ff-48ad-bcbc-4bbb708b01cb)